packages/ui/src/models/camel/camel-k-resource.ts

```diff
@@ -25,14 +26,19 @@ export const CAMEL_K_K8S_API_VERSION_V1 = 'camel.apache.org/v1';
 
 export abstract class CamelKResource implements CamelResource {
   static readonly PARAMETERS_ORDER = ['apiVersion', 'kind', 'metadata', 'spec', 'source', 'steps', 'sink'];
+  // static serializer = new YamlResourceSerializer();
   readonly sortFn = createCamelPropertiesSorter(CamelKResource.PARAMETERS_ORDER) as (a: unknown, b: unknown) => number;
   protected resource: CamelKType;
   private metadata?: MetadataEntity;
-  private comments: string[] = [];
 
-  constructor(resource?: CamelKType) {
-    if (resource) {
-      this.resource = resource;
+  constructor(
+    parsedResource: unknown,
+    private readonly serializer: CamelResourceSerializer = new YamlCamelResourceSerializer(),
```

Comment:
Perhaps we could set it as `protected` to perform some tests if we want to.
```suggestion
    protected readonly serializer: CamelResourceSerializer = new YamlCamelResourceSerializer(),
```

Reply:
important :point_up: